validator/coordinator/src/main/java/tech/pegasys/teku/validator/coordinator/BlockFactory.java

```diff
@@ -87,7 +94,12 @@ public BeaconBlock createUnsignedBlock(
     }
 
     // Collect attestations to include
-    final BeaconState blockSlotState = stateTransition.process_slots(blockPreState, newSlot);
+    final BeaconState blockSlotState;
+    if (maybeBlockSlotState.isPresent()) {
+      blockSlotState = maybeBlockSlotState.get();
+    } else {
+      blockSlotState = stateTransition.process_slots(blockPreState, newSlot);
+    }
```

Comment:
might be able to use `orElseGet` instead here?

Reply:
ah I remember trying that before now.... `process_slots` can throw checked exceptions so it doesn't work well with `orElseGet`.